File: python/maximumsubarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        curmax = maxsum = nums[0]
        for i in nums[1:]:
            curmax = max(i, curmax+i)
            maxsum = max(maxsum, curmax)
        return maxsum

File: java/MaximumSubarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int maxSub = Integer.MIN_VALUE;
        int curMax = 0;
        for (int i = 0; i < nums.length; i++) {
            curMax = (curMax + nums[i]) > nums[i] ? curMax + nums[i] : nums[i];
            maxSub = Math.max(maxSub, curMax);
        }
        return maxSub;
    }
}
